### DISCUSSION TITLE
Transmission "HOST_DOWNLOAD_PATH" in docker-compose.transmission-vpn.yml

### DISCUSSION THREAD

**Discussion by jpbaril** [lardbit/nefarious#145] [Q&A] [answered]
Is it normal that in https://github.com/lardbit/nefarious/blob/71b4728c535b4ccde20326850f781047c3d83956/docker-compose.transmission-vpn.yml#L43 it shows "/data" as the "HOST_DOWNLOAD_PATH" value but that in the other hand in https://github.com/lardbit/nefarious/blob/71b4728c535b4ccde20326850f781047c3d83956/docker-compose.base.yml#L85 it's "/downloads" ?

That file https://github.com/lardbit/nefarious/blob/71b4728c535b4ccde20326850f781047c3d83956/transmission-settings.json#L2 also shows  "/downloads" as the "download-dir".

I tried to _correct_ it my docker-compose.transmission-vpn.yml but I'm not sure as it's said to not modify docker-compose files.

Also, is it related to my change that variable values in my transmission-settings.json file are not taken into account by Transmission?

Thank you.

**Comment by lardbit** [comment, marked as answer]
Hi @jpbaril,

The reason the `HOST_DOWNLOAD_PATH` differs in those two files is because they are different *docker images* which expects the downloads to live in different places.  The main `docker-compose.yml` file uses only [transmission](https://github.com/linuxserver/docker-transmission) and the `docker-compose.transmission-vpn.yml` uses [transmission+vpn](https://github.com/haugene/docker-transmission-openvpn/).  Each docker image is unique and expects different parameters, but **you'll only be using one of them**, depending if you want a VPN baked in or not.

The `transmission-settings.json` file is for the regular transmission (without vpn) and that's why it's download path is `/downloads` just as in `docker-compose.yml`.  Editing the file will have no effect on the vpn compose file (`docker-compose.transmission-vpn.yml`).  

You are correct in that you shouldn't need to edit any of the docker-compose.yml files as settings are meant to be configured through the `.env` file.  

Let me know if you think the instructions are missing anything to make it more clear.

**Reply by jpbaril** [reply]
But if I want to:
- use "/downloads" for HOST_DOWNLOAD_PATH volume mapping (I want to separate Transmission home and download dir)
- add another volume for "TRANSMISSION-HOME" path so that config changes done through Transmission UI are persistent
- open "8118" port for the http proxy included in transmission+vpn

is it then ok to modify the docker-compose.transmission-vpn.yml? or should I use a third extending .yml file? or is there a way to do all that in the .env file?

I have some degree of knowledge of Docker but this particular complex setup is a bit beyond.
Thanks again

**Comment by lardbit** [comment]
If you're using transmission + vpn, then you'll only be using the `docker-compose.transmission-vpn` which uses the [transmission-openvpn](https://haugene.github.io/docker-transmission-openvpn/run-container/) image and unfortunately doesn't have the ability to separate the downloads from the transmission settings.  You can read it's instructions on that link.  It puts everything in the container's `/data` directory.

If you want to use haugene's proxy, then you're right in that you'll want to edit the `docker-compose.transmission-vpn.yml`.  My "*don't edit the compose files*" warning is for general users so they don't accidentally break something.  

I hope that answers your questions.
